typescript/sdk/src/ism/HyperlaneIsmFactory.ts

```diff
@@ -76,11 +78,13 @@ export class HyperlaneIsmFactory extends HyperlaneApp<ProxyFactoryFactories> {
     );
   }
 
-  async deploy<C extends IsmConfig>(
-    chain: ChainName,
-    config: C,
-    origin?: ChainName,
-  ): Promise<DeployedIsm> {
+  async deploy<C extends IsmConfig>(params: {
+    chain: ChainName;
+    config: C;
+    origin?: ChainName;
+    mailbox?: Address;
+  }): Promise<DeployedIsm> {
```

Comment:
now that we have the two `ModuleType` and `IsmType` we can probably return a generic contract so we don't have to cast eg `as OpStackIsm`